aptos-move/move-examples/marketplace/sources/coin_listing.move

```diff
@@ -0,0 +1,1080 @@
+        coin::deposit(fee_schedule::fee_address(fee_schedule), commission);
+
+        if (royalty_charge != 0) {
+            let royalty = coin::extract(&mut coins, royalty_charge);
+            coin::deposit(royalty_addr, royalty);
+        };
+
+        coin::deposit(seller, coins);
+
+        let purchase_event_data = PurchaseEvent {
+            purchaser: purchaser_addr,
+            price,
+            commission: commission_charge,
+            royalties: royalty_charge,
+        };
+        event::emit_event(&mut purchase_event, purchase_event_data);
+        event::destroy_handle(purchase_event);
+    }
+
+    /// End a fixed price listing early.
+    public entry fun end_fixed_price<CoinType>(
+        seller: &signer,
+        object: Object<Listing>,
+    ) acquires FixedPriceListing {
+        let expected_seller_addr = signer::address_of(seller);
+        let (actual_seller_addr, _fee_schedule) = listing::complete(object, expected_seller_addr);
+        assert!(expected_seller_addr == actual_seller_addr, error::permission_denied(ENOT_SELLER));
+
+        let listing_addr = object::object_address(&object);
+        assert!(exists<FixedPriceListing<CoinType>>(listing_addr), error::not_found(ENO_LISTING));
+        let FixedPriceListing {
+            price: _,
+            purchase_event,
+        } = move_from<FixedPriceListing<CoinType>>(listing_addr);
+
+        let purchase_event_data = PurchaseEvent {
+            purchaser: expected_seller_addr,
+            price: 0,
+            commission: 0,
+            royalties: 0,
+        };
+        event::emit_event(&mut purchase_event, purchase_event_data);
+        event::destroy_handle(purchase_event);
+    }
+
+    /// Make a bid on a listing. If the listing comes in near the end of an auction, the auction
+    /// may be extended to give at least minimum_bid_time_before_end time remaining in the auction.
+    public entry fun bid<CoinType>(
+        bidder: &signer,
+        object: Object<Listing>,
+        bid_amount: u64,
+    ) acquires AuctionListing {
+        let listing_addr = listing::assert_started(&object);
+        assert!(exists<AuctionListing<CoinType>>(listing_addr), error::not_found(ENO_LISTING));
+        let auction_listing = borrow_global_mut<AuctionListing<CoinType>>(listing_addr);
+
+        let now = timestamp::now_seconds();
+        assert!(now < auction_listing.auction_end_time, error::invalid_state(EAUCTION_ENDED));
+
+        let (previous_bidder, previous_bid, minimum_bid) = if (option::is_some(&auction_listing.current_bid)) {
+            let Bid { bidder, coins } = option::extract(&mut auction_listing.current_bid);
+            let current_bid = coin::value(&coins);
+            coin::deposit(bidder, coins);
+            (option::some(bidder), option::some(current_bid), current_bid + auction_listing.bid_increment)
+        } else {
+            (option::none(), option::none(), auction_listing.starting_bid)
```

Comment:
I think having option::none everywhere makes the code more complex and less readable. Wdyt about using 0 to indicate there's been no bid?

Reply:
May be. I don't feel strongly here though